ksql-parser/src/main/java/io/confluent/ksql/parser/tree/CreateStreamAsSelect.java

```diff
@@ -39,33 +37,25 @@ public CreateStreamAsSelect(
       final QualifiedName name,
       final Query query,
       final boolean notExists,
-      final Map<String, Literal> properties,
+      final CreateSourceAsProperties properties,
       final Optional<Expression> partitionByColumn) {
     super(location, name, query, notExists, properties, partitionByColumn);
   }
 
   private CreateStreamAsSelect(
       final CreateStreamAsSelect other,
-      final Map<String, Literal> properties
+      final CreateSourceAsProperties properties
   ) {
     super(other, properties);
   }
 
   @Override
   public Sink getSink() {
-    final Map<String, Expression> sinkProperties = getPartitionByColumn()
-        .map(exp -> (Map<String, Expression>)ImmutableMap.<String, Expression>builder()
-            .putAll(getProperties())
-            .put(DdlConfig.PARTITION_BY_PROPERTY, exp)
```

Comment:
Rather than merging `PARTITION_BY_PROPERTY` into the properties, its now just passed as its own field.